src/main/scala/rps/GameRepository.scala

```diff
@@ -0,0 +1,30 @@
+package rps
+
+import Move._
+import Result._
+import Match._
+import scala.collection.concurrent.TrieMap
+
+trait GameRepository {
+
+  def saveGameOutcome(
+      theMatch: Match
+  ): Unit;
+
+  def loadGameOutcome(): Option[Match];
```

Comment:
Just a suggestion, the repository could be more generic and allow to store/load any game, and the service could determine which name to use, instead of "hardcoding" the last one in the repository.

Reply:
Gotcha! In any case, according to Step 7 specs, the service should return only the last available game. If this is correct, the modifications to be done here are just to introduce a parameter in "loadGameOutcome" (e.g. a string or a uuid) and let the service decide which string or uuid use to store the last available game. Is this correct?

Reply:
Yes, something like that.

I'd like to keep the repository as agnostic as possible and encode the "last game" logic in the service.

Reply:
Perfect! Check the latest modifications then.